### PR TITLE
Make ADD untar tar.gz files

### LINKED ISSUE #298: ADD Dockerfile command does not unpack tarballs
**Describe the bug**
`ADD tarball.tar.gz /` is supposed to unpack the said tarball into the root. Instead, it just copies it into it.

**To Reproduce**
The specific Dockerfile I ran into this at:
```
FROM scratch
    
ENV LANG=C.UTF-8
ENV LC_ALL=C.UTF-8

ARG DEBIAN_FRONTEND=noninteractive
  
ARG OSPACK=debian.tar.gz
ADD ${OSPACK} /

RUN apt-get update
```
Here, `debian.tar.gz` is a minimal Debian rootfs tarball.

**Expected behavior**
The tarball is unpacked, so that I can run commands using its contents.

Instead, I’m getting:
```
info	* Step 6/7 (modifyfs) : ADD debian.tar.gz /  (465a0d0c)
info	* Execute ADD debian.tar.gz /  (465a0d0c) took 91.261753ms
info	* Not committing step ADD debian.tar.gz /  (465a0d0c)
info	* Step 6/7 (modifyfs) : RUN apt-get update  (d6692f4c)
error	failed to execute build plan: execute stage: build stage 0: build node: do execute: execute step: cmd start: exec: "sh": executable file not found in $PATH
```

### PR DESCRIPTION
**Pull request by yiranwang52**
Fix #298

### CODE REVIEW

File: lib/builder/step/add_copy_step.go [hunk 1]
```diff
@@ -58,12 +58,13 @@ type addCopyStep struct {
 	toPath        string
 	chown         string
 	preserveOwner bool
+	isAdd         bool
```
Comment (XavierCheng1996) [inline lib/builder/step/add_copy_step.go]: is this similar to `fromAdd`?

File: lib/snapshot/copy_op_test.go [hunk 2]
```diff
@@ -15,13 +15,16 @@
...
 	"fmt"
 	"io/ioutil"
 	"os"
+	"path"
 	"path/filepath"
 	"testing"
 
 	"github.com/uber/makisu/lib/pathutils"
+	"github.com/uber/makisu/lib/tario"
 	"github.com/uber/makisu/lib/utils/testutil"
```
Comment (XavierCheng1996) [inline lib/snapshot/copy_op_test.go]: could we make it just two blocks?

File: lib/builder/step/add_copy_step.go (hunk 1 above)
Comment (yiranwang52) [reply lib/builder/step/add_copy_step.go]: yes

File: lib/snapshot/copy_op_test.go (hunk 2 above)
Comment (yiranwang52) [reply lib/snapshot/copy_op_test.go]: it's two blocks.

File: lib/builder/step/add_copy_step.go (hunk 1 above)
Comment (yiranwang52) [reply lib/builder/step/add_copy_step.go]: I intentionally made them different.

**Pull request merged**